### DISCUSSION TITLE
Integration with consul

### DISCUSSION THREAD

**Discussion by tzahimizrahi** [Kong/kong#7279] [Ideas and feature requests]
I'll give a shoot here. 
Is there a future plan to add Integration with Consul service discovery?

**Comment by Tieske** [comment]
There has been an occasional request, but since consul exposes a dns interface, that is generally used.

**Comment by tzahimizrahi** [comment]
Consul dns response ttl is 0 by default, that means Kong will have to query every time it has a request which will cause high latency.

**Comment by Tieske** [comment]
That is a logical consequence. I'd advise not to use `ttl == 0` with Kong, a low number (few seconds) should be fine since Kong will resolve in the background. See also the [`dns_stale_ttl` property](https://docs.konghq.com/1.0.x/configuration/#dns_stale_ttl).

**Comment by subnetmarco** [comment]
@tzahimizrahi does this answer your question or did you have something else in mind?

**Comment by tzahimizrahi** [comment]
@subnetmarco DNS resolution is good however I hoped to find an integration with Consul service catalog which is a more reliable and relevant method for high rate scaling services echo systems

**Comment by subnetmarco** [comment]
@tzahimizrahi the request would be to integrate with their [Catalog HTTP API](https://www.consul.io/api/catalog.html) in order to [list nodes for a service](https://www.consul.io/api/catalog.html#list-nodes-for-service) and proxy to those nodes? Could you please describe the behavior you would like to see in Kong?

**Comment by tzahimizrahi** [comment]
@subnetmarco right. the implementation should include a watch on consul services catalog that trigger an update Kong beckend upstream according to a service node health change.
Tnx!

**Comment by ccl0326** [comment]
@subnetmarco there is some limited in consul dns interface, the service and tag names in consul should't allow some characters like dot.

FYI. 
https://github.com/hashicorp/consul/issues/215
https://www.consul.io/docs/agent/services.html#service-and-tag-names-with-dns
